File: apps/processes/src/analysis/ogc-conformance.step.ts
import type { ApiRouteConfig, Handlers } from "motia";
import { z } from "zod";

/**
 * OGC API Processes - Conformance
 * GET /conformance
 *
 * Returns list of conformance classes the API implements.
 */

const PROCESSES_CONFORMANCE = [
  "http://www.opengis.net/spec/ogcapi-processes-1/1.0/conf/core",
  "http://www.opengis.net/spec/ogcapi-processes-1/1.0/conf/ogc-process-description",
  "http://www.opengis.net/spec/ogcapi-processes-1/1.0/conf/json",
  "http://www.opengis.net/spec/ogcapi-processes-1/1.0/conf/job-list",
  "http://www.opengis.net/spec/ogcapi-processes-1/1.0/conf/dismiss",
];

export const config: ApiRouteConfig = {
  name: "OGCConformance",
  type: "api",
  path: "/conformance",
  method: "GET",
  description: "OGC API Processes conformance classes",
  emits: [],
  responseSchema: {
    200: z.object({
      conformsTo: z.array(z.string()),
    }),
  },
};

export const handler: Handlers["OGCConformance"] = async (_req, { logger }) => {
  logger.info("OGC Conformance requested");

  return {
    status: 200,
    body: {
      conformsTo: PROCESSES_CONFORMANCE,
    },
  };
};

File: apps/processes/src/analysis/ogc_conformance_step.py
"""
OGC API Processes - Conformance
GET /conformance

Returns list of conformance classes the API implements.
"""

import sys

# Add paths before any lib imports
for path in ["/app/apps/processes/src", "/app/apps/core/src", "/app/packages/python/goatlib/src"]:
    if path not in sys.path:
        sys.path.insert(0, path)

from lib.ogc_base import PROCESSES_CONFORMANCE, pydantic_response
from lib.ogc_schemas import Conformance

config = {
    "name": "OGCConformance",
    "type": "api",
    "path": "/conformance",
    "method": "GET",
    "description": "OGC API Processes conformance classes",
    "emits": [],
}


async def handler(req, context):
    """Handle GET /conformance request."""
    context.logger.info("OGC Conformance requested")

    conformance = Conformance(conformsTo=PROCESSES_CONFORMANCE)

    return pydantic_response(conformance)
